CMakeLists.txt

```diff
@@ -231,6 +240,18 @@ cprover_default_properties(
     xml)
 
 option(WITH_JBMC "Build the JBMC Java front-end" ON)
+
+# Don't build jbmc on MacOS running on ARM
+# See https://github.com/Homebrew/homebrew-core/blob/master/Formula/cbmc.rb
+# See https://github.com/diffblue/cbmc/issues/6343
+execute_process(COMMAND uname -p
```

Comment:
I don't really think that's needed at all - I'm able to build on my ARM Mac without this flag on, cleanly.

I believe this may be an artefact on the home-brew formula since the early days of `M1`, where various environments (including `java`) had issues with their support.

My local machine (Apple Silicon ARM) is reporting the following if it helps:

```sh
$ fotiskoutoulakis@Fotiss-Mac-mini build_develop % ./bin/jbmc
JBMC version 5.38.0 (cbmc-5.38.0-7-gf0ab57e29a) 64-bit arm64 macos
Please give exactly one class name, and/or use -jar jarfile or --gb goto-binary
$ fotiskoutoulakis@Fotiss-Mac-mini build_develop % java --version
java 16.0.1 2021-04-20
Java(TM) SE Runtime Environment (build 16.0.1+9-24)
Java HotSpot(TM) 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)
```

Reply:
@NlightNFotis Might be worth checking if that Java 16 you have installed is a x86 or arm64 build... but either way, I think this reinforces that its not specifically a macOS M1 issue - its a Java 16/17 issue.

Reply:
@chrisr-diffblue Yeah, I think you're right - on my system, it looks like java is a fat binary of two architectures:

```sh
$ fotiskoutoulakis@Fotiss-Mac-mini build_develop % cat `which java` | file -
/dev/stdin: Mach-O universal binary with 2 architectures: [x86_64:Mach-O 64-bit executable x86_64] [arm64e:Mach-O 64-bit executable arm64e]
```